pal/ios-osx/tlsio_appleios.c

```diff
@@ -409,6 +409,36 @@ static int tlsio_appleios_close_async(CONCRETE_IO_HANDLE tls_io, ON_IO_CLOSE_COM
     return result;
 }
 
+char* CFStringCopyUTF8String(CFStringRef aString) {
+    if (aString == NULL)
+    {
+        return NULL;
+    }
+
+    CFIndex length = CFStringGetLength(aString);
+    CFIndex maxSize = CFStringGetMaximumSizeForEncoding(length, kCFStringEncodingUTF8) + 1;
+    char *buffer = (char *)malloc(maxSize);
+    if (CFStringGetCString(aString, buffer, maxSize, kCFStringEncodingUTF8))
+    {
+        return buffer;
+    }
+    free(buffer); // If we failed
+    return NULL;
+}
+
+void LogAndReleaseCFError(CFErrorRef streamError)
+{
+    if (streamError != NULL)
+    {
+        CFStringRef errCFStr = CFCopyDescription(streamError);
+        char* errStr = CFStringCopyUTF8String(errCFStr);
+        LogError("%s: Network Error %s", __FUNCTION__, errStr);
```

Comment:
>errStr [](start = 55, length = 6)

this will just print NULL in case errStr is NULL?

Reply:
yes.